workers/unity/Assets/Fps/Scripts/SetupLogic/SimulatedPlayerCoordinatorWorkerConnector.cs

```diff
@@ -1,209 +1,339 @@
-    private readonly List<EntityId> localSimulatedPlayers = new List<EntityId>();
+        private const string DeploymentNameFlag = "deploymentName";
 
-    private readonly List<GameObject> simulatedPlayerConnectors = new List<GameObject>();
+        public GameObject SimulatedPlayerWorkerConnector;
 
-    protected override async void Start()
-    {
-        Application.targetFrameRate = 60;
-        await AttemptConnect();
-    }
+        // TODO: Nice Unity tooltips
+        public int MaxSimulatedPlayerCount = 1;
+        public int SimulatedPlayerCreationInterval = 5;
+        public bool UseSessionFlow;
 
-    protected override string GetWorkerType()
-    {
-        return WorkerUtils.SimulatedPlayerCoordinator;
-    }
+        private readonly Dictionary<EntityId, List<GameObject>> proxies = new Dictionary<EntityId, List<GameObject>>();
+        private readonly List<EntityId> localSimulatedPlayers = new List<EntityId>();
 
-    protected override async void HandleWorkerConnectionEstablished()
-    {
-        base.HandleWorkerConnectionEstablished();
+        private readonly List<GameObject> simulatedPlayerConnectors = new List<GameObject>();
 
-        Worker.World.GetOrCreateSystem<MetricSendSystem>();
+        private bool connectPlayersWithDevAuth;
 
-        CheckWorkerFlags();
+        private CancellationTokenSource tokenSource;
 
-        if (SimulatedPlayerWorkerConnector != null)
+        protected override async void Start()
         {
-            while (Application.isPlaying && simulatedPlayerConnectors.Count < DefaultSimulatedPlayerCount)
-            {
-                var simulatedPlayer = Instantiate(SimulatedPlayerWorkerConnector, transform.position, transform.rotation);
-                var simulatedPlayerConnector = simulatedPlayer.GetComponent<SimulatedPlayerWorkerConnector>();
-                await simulatedPlayerConnector
-                    .ConnectSimulatedPlayer(Worker.LogDispatcher, SimulatedPlayerDevAuthTokenId, SimulatedPlayerTargetDeployment);
-                simulatedPlayerConnector.SpawnPlayer(simulatedPlayerConnectors.Count);
+            var args = CommandLineArgs.FromCommandLine();
 
-                simulatedPlayerConnectors.Add(simulatedPlayer);
-                await Task.Delay(TimeSpan.FromSeconds(
-                    Random.Range(DefaultSimulatedPlayerCreationInterval, 1.25f * DefaultSimulatedPlayerCreationInterval)));
-            }
+            string deploymentName = string.Empty;
 
-            if (Application.isPlaying)
+            if (args.TryGetCommandLineValue(DeploymentNameFlag, ref deploymentName))
             {
-                StartCoroutine(MonitorSimulatedPlayers());
+                connectPlayersWithDevAuth = deploymentName != "local";
             }
+            else
+            {
+                // We are probably in the Editor.
+                connectPlayersWithDevAuth = false;
+            }
+
+            Application.targetFrameRate = 60;
```

Comment:
Open a ticket for making these configurable

Reply:
https://improbableio.atlassian.net/browse/UTY-2115